### DISCUSSION TITLE
FreeBSD support

### DISCUSSION THREAD

**Discussion by h-2** [filips123/PWAsForFirefox#419] [Q&A] [answered]
Hi! Does this work on FreeBSD or is support planned?

**Comment by filips123** [comment]
It's currently not supported because the system integration is specific for Windows, macOS and Linux. However, if BSD supports XDG [Desktop Entry Specification](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html) (which is used on Linux), it should be pretty easy to add support for it (except automatic runtime installation, as Mozilla does not provide official BSD builds).

However, it will probably require manual installation from source, as I don't know how BSD package managers work.

**Comment by filips123** [comment, marked as answer]
Support for BSDs (DragonFlyBSD, FreeBSD, OpenBSD, NetBSD, and derivates) will be added in the next release. It relies on XDG Desktop Entry Specification (made for Linux), so I think it should work on most "normal" desktop environments. However, note that it's not regularly tested on BSD.

As Mozilla does not provide official pre-built Firefox for BSD, the runtime needs to be manually installed. This means you'll need to manually copy Firefox installation from `/usr/local/lib/firefox` (or the appropriate location where Firefox is installed) to [the correct directory](https://pwasforfirefox.filips.si/resources/installation-directories/#runtime) (for user data, it's the same as for Linux).

You will need to build the project from source. Instructions for this are similar to manual building on other platforms (using `cargo-make`), however, before running `makers`, you'll need to replace `/usr/` with `/usr/local/` inside `Makefile.toml`'s `[tasks.install.linux]` section. (This step will probably be improved in the future once `cargo-make` adds better support for BSDs).

**Reply by filips123** [reply]
I now released 2.9.0 with support for BSD. Please let me know if you have any problems.

**Reply by h-2** [reply]
Awesome, I'll give it a try!

**Reply by filips123** [reply]
`cargo-make` has added support for better OS detection, so it should now be possible to install PWAsForFirefox on BSD simply with `makers install`.
